sherpa/astro/io/__init__.py

```diff
@@ -948,6 +1066,463 @@ def convert(column, dtype):
+    """Extract FITS column and header information.
+
+    Unlike the other pack routines this returns data for
+    multiple blocks. At present this ignores the ORDER column
+    in the MATRIX block.
+
+    Notes
+    -----
+    The `RMF Data Extension header page
+    <https://heasarc.gsfc.nasa.gov/docs/heasarc/caldb/docs/memos/cal_gen_92_002/cal_gen_92_002.html>`_
+    lists the following keywords as either required or
+    we-really-want-them. First for the MATRIX block:
+
+        EXTNAME = 'MATRIX' or 'SPECRESP MATRIX'
+        TELESCOP - the "telescope" (ie mission/satellite name).
+        INSTRUME - the instrument/detector.
+        FILTER - the instrument filter in use (if any)
+        CHANTYPE - PI or PHA
+        DETCHANS - the total number of raw detector PHA channels
+        HDUCLASS = 'OGIP' - file format is OGIP standard.
+        HDUCLAS1 = 'RESPONSE' - extension contains response data.
+        HDUCLAS2 = 'RSP_MATRIX' - extension contains a response matrix.
+        HDUVERS = '1.3.0' - version of the file format.
+        TLMIN# - the first channel in the response.
+
+    These are optional but mechanical so can be added:
+
+        NUMGRP - sum(N_GRP)
+        NUMELT - sum(N_CHAN)
+
+    These are optional:
+
+        LO_THRES - minimum probability threshold
+        HDUCLAS3 - 'REDIST', 'DETECTOR', 'FULL'
+
+    For the EBOUNDS extension we have:
+
+        EXTNAME  = 'EBOUNDS'
+        TELESCOP
+        INSTRUME
+        FILTER
+        CHANTYPE
+        DETCHANS
+        HDUCLASS = 'OGIP'
+        HDUCLAS1 = 'RESPONSE'
+        HDUCLAS2 = 'EBOUNDS'
+        HDUVERS = '1.2.0'
+
+    We do not add the RMFVERSN, HDUVERS1, or HDUVERS2 keywords which
+    are marked as obsolete.
+
+    """
+
+    # The UI layer wraps up a DataRMF into RMF1D quite often, so
+    # support both types as input.
+    #
+    if not isinstance(dataset, (DataRMF, RMF1D)):
+        raise IOErr("data set is not a RMF")
+
+    rmfdata = _reconstruct_rmf(dataset)
+
+    # The default keywords; these wil be over-ridden by
+    # anything set by the input.
+    #
+    default_matrix_header = {
+        "EXTNAME": "MATRIX",
+        "HDUCLASS": "OGIP",
+        "HDUCLAS1": "RESPONSE",
+        "HDUCLAS2": "RSP_MATRIX",
+        "HDUVERS": "1.3.0",
+        "TELESCOP": "none",
+        "INSTRUME": "none",
+        "FILTER": "none",
+        "CHANTYPE": "PI",
+        "DETCHANS": "none",
+        "NUMGRP": 0,
+        "NUMELT": 0
+    }
+
+    default_ebounds_header = {
+        "EXTNAME": "EBOUNDS",
+        "HDUCLASS": "OGIP",
+        "HDUCLAS1": "RESPONSE",
+        "HDUCLAS2": "EBOUNDS",
+        "HDUVERS": "1.2.0",
+        "TELESCOP": "none",
+        "INSTRUME": "none",
+        "FILTER": "none",
+        "CHANTYPE": "PI",
+        "DETCHANS": "none"
+    }
+
+    # Header Keys
+    header = dataset.header
+
+    # Merge the keywords (at least for the MATRIX block).
+    #
+    matrix_header = {**default_matrix_header, **header}
```

Comment:
optional. Either works...
```suggestion
    matrix_header = default_matrix_header | header
```

Reply:
I thought the `|` syntax was a python 3.10-ism but I see from https://docs.python.org/3/library/stdtypes.html#typesmapping it's python 3.9. Which we can assume, so that's nice.